lib/Frontend/FrontendInputsAndOutputs.cpp

```diff
@@ -303,6 +308,18 @@ void FrontendInputsAndOutputs::setMainAndSupplementaryOutputs(
       AllInputs[i].setOutputFilename(outputFiles[i]);
   }
   SupplementaryOutputs = supplementaryOutputs;
+
+  if (hasInputs() && primaryInputCount() < 2) {
```

Comment:
I feel like it's generally weird to coalesce "whole-module" and "only one primary" like this. I'd prefer if you just ask for both of those conditions separately.

Reply:
Changed to `hasUniquePrimaryInput() || (hasInputs() && isWholeModule())`

Reply:
Tx